lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -39,9 +39,17 @@ final class ViewFocusBinding {
     }
   }
 
+  late final DomEventListener _handleFocusin = createDomEventListener(
+    (DomEvent event) => _handleFocusChange(event.target as DomElement?),
+  );
+
+  late final DomEventListener _handleFocusout = createDomEventListener(
+    (DomEvent event) => _handleFocusChange(event.relatedTarget as DomElement?),
+  );
```

Comment:
Reminds me of [something](https://github.com/flutter/engine/pull/50177#discussion_r1480875921)... :P

Reply:
Great catch. No wonder you're a true pro!

Reply:
Will elab in the comment below on why this change is needed.